Added most native Windows dependencies to the fetch script

Mailaender opened the pull request:
Further work upon https://github.com/OpenRA/OpenRA/issues/6863 to make it possible to package this on Linux again. Otherwise it will unnecessarily fail with:

```
Fetching SDL2 from nuget
./fetch-thirdparty-deps.sh: line 51: nuget: command not found
```

even though all CLI dependencies are resolvable.

Mailaender committed:
don't fetch native Windows binaries on Unix

Mailaender committed:
fetch Lua 5.1 from nuget.org

Mailaender committed:
fetch Zlib from nuget

Mailaender committed:
create a thirdparty/windows directory if it does not exit

Phrohdoh commented:
I don't understand how we will build the Windows packages with these changes.

A bit off topic: Why do we && on commands that won't fail instead of pushd and popd?

Mailaender commented:
> I don't understand how we will build the Windows packages with these changes.

`make windows-dependencies` but that will be changed by https://github.com/OpenRA/OpenRA/pull/7345 soon.

> A bit off topic: Why do we && on commands that won't fail instead of pushd and popd?

I am not the expert here. All I know is that Makefile syntax is not exactly like shell. Have a look at https://www.gnu.org/software/make/manual/make.html

Phrohdoh commented:
Thanks!

pchote commented on thirdparty/fetch-thirdparty-deps-windows.sh:
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# Die on any error for Travis CI to automatically retry:
```
We should merge #7345 first, then this can be removed.

Mailaender replied:
This is still useful.

Mailaender replied:
Actually, the other way round. This should get merged first as https://github.com/OpenRA/OpenRA/pull/7345 has to adapt to the changes here.

ScottNZ commented on thirdparty/fetch-thirdparty-deps.ps1:
```diff
@@ -1,3 +1,5 @@
+New-Item -ItemType Directory -Force -Path windows
```
This should use the alias `mkdir`

Mailaender replied:
Can you give me the full line? I can't test myself for the correct syntax and AppVeyor is slow to respond.

ScottNZ replied:
```
mkdir windows -Force >$null
```

Mailaender replied:
Thanks, updated.

ScottNZ commented on thirdparty/fetch-thirdparty-deps.ps1:
```diff
@@ -69,5 +71,29 @@ if (!(Test-Path "Mono.Nat.dll"))
+if (!(Test-Path "windows/zlib1.dll"))
+{
+	echo "Fetching ZLib from NuGet."
+	nuget install zlib.redist -Version 1.2.8.7
```
This should have `./nuget.exe`

ScottNZ commented on thirdparty/fetch-thirdparty-deps.ps1:
```diff
@@ -69,5 +71,29 @@ if (!(Test-Path "Mono.Nat.dll"))
+if (!(Test-Path "windows/soft_oal.dll"))
+{
+	echo "Fetching ZLib from NuGet."
+	nuget install openal.redist -Version 2.0.7
```
This should have `./nuget.exe`

Mailaender commented:
Updated.

LipkeGu commented:
:+1:

Mailaender commented:
Removed https://www.nuget.org/packages/openal.redist/ for now as we switched to the incompatible and unavailable http://kcat.strangesoft.net/openal.html#download which is not yet packaged.

LipkeGu commented:
now it works :)

LipkeGu commented:
:+1:

ScottNZ commented:
:+1:
